src/cdomains/floatDomain.ml

```diff
@@ -186,7 +186,22 @@ module FloatInterval = struct
   (** evaluation of the binary operations *)
   let eval_binop eval_operation op1 op2 =
     norm @@ match (op1, op2) with 
-    | Some v1, Some v2 -> eval_operation v1 v2
+    | Some v1, Some v2 -> 
+      let is_exact (lower, upper) = Float.compare lower upper == 0 in
+      let is_exact_before = is_exact v1 && is_exact v2 in
+      let result = eval_operation v1 v2
+      in if is_exact_before then 
```

Comment:
Pull this is into the inner `if` making it `not is_exact_after && is_exact_before` and remove the outer one. You could even avoid the match by using `Option.iter`